### PR TITLE
Do not use --set-gtid-purged parameter on MariaDB

### LINKED ISSUE #40: Protector does not work with MariaDB
Protector sets a `--set-gtid-purged=OFF` parameter when dumping that does not work with the MariaDB version.

### PR DESCRIPTION
**Pull request by lupinitylabs**
This removes the use of the unsupported --set-gtid-purged parameter on MariaDB systems.
Closes #40.

### CODE REVIEW

**Comment by lupinitylabs** [comment]
Was tested on a MariaDB production server: works fine.

File: src/Protector.php [hunk 1]
```diff
@@ -368,7 +369,10 @@ protected function generateDump(string $destinationFilePath, array $options = []
         $dumpOptions->push(sprintf('-p%s', escapeshellarg($this->connectionConfig['password'])));
         $dumpOptions->push(sprintf('--max-allowed-packet=%s', escapeshellarg(config('protector.maxPacketLength'))));
         $dumpOptions->push('--no-create-db');
-        $dumpOptions->push('--set-gtid-purged=off');
+
+        if (! DB::connection()->isMaria()) {
```
Comment (gael-connan-cybex) [inline src/Protector.php:373]: style

File: src/Protector.php [hunk 2]
```diff
@@ -368,7 +369,10 @@ protected function generateDump(string $destinationFilePath, array $options = []
...
+            $dumpOptions->push('--set-gtid-purged=off');
+        }
```
Comment (gael-connan-cybex) [inline src/Protector.php:375]: looks great, while youre at it, can you write a line or two into the readme file, regarding the supported database software? preferrably near the top.

File: src/Protector.php (hunk 1 above)
Comment (lupinitylabs) [reply src/Protector.php]: fixed

File: src/Protector.php (hunk 2 above)
Comment (lupinitylabs) [reply src/Protector.php:384]: Done.

File: src/Protector.php [hunk 3]
```diff
@@ -385,7 +386,10 @@ protected function generateDump(array $options = []): ?string
         $dumpOptions->push(sprintf('-p%s', escapeshellarg($this->connectionConfig['password'])));
         $dumpOptions->push(sprintf('--max-allowed-packet=%s', escapeshellarg(config('protector.maxPacketLength'))));
         $dumpOptions->push('--no-create-db');
-        $dumpOptions->push('--set-gtid-purged=off');
+
+        if (!DB::connection()->isMaria()) {
```
Comment (gael-connan-cybex) [inline src/Protector.php:390]: for some reason, this wonderful little line of code leads to the following failure codes in docker

SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for laravel-protector-mysql_testing-1 failed: Name or service not known

SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for laravel-protector-mysql_testing-1 failed: Temporary failure in name resolution

and sometimes

SQLSTATE[HY000] [2002] Connection refused

Comment (lupinitylabs) [reply src/Protector.php:390]: As we have already established, the network connection is not properly set up within the package, so it is probably a configuration error. You probably have to work that out with Jörn if you cannot solve it yourself.

Comment (gael-connan-cybex) [reply src/Protector.php:390]: it simply works when i comment your line of code

Comment (lupinitylabs) [reply src/Protector.php:390]: That's not really a solution, right?

Comment (gael-connan-cybex) [reply src/Protector.php:390]: didnt mean to imply that

**Review by gael-connan-cybex — APPROVED** [review]
tested

File: src/Protector.php (hunk 3 above)
Comment (holyfabi) [reply src/Protector.php]: We now make sure we are using the correct database by specifying $this->connection when we check that the connected database is MariaDB.

**Pull request merged**